src/utils_lua.c

```diff
@@ -274,38 +235,200 @@ int luaC_pushvalue(lua_State *L, value_t v, int ds_type) /* {{{ */
   return 0;
 } /* }}} int luaC_pushvalue */
 
-int luaC_pushvaluelist(lua_State *L, const data_set_t *ds,
-                       const value_list_t *vl) /* {{{ */
+int luaC_pushlabelset(lua_State *L, label_set_t label) /* {{{ */
 {
+  /*
+    label_set_t will be mapped to the following Lua table:
+
+    label = {
+      [1] => { name => value },
+      ...
+      [N] => { ... }
+    }
```

Comment:
Why did you chose this approach? I'd find the following more intuitive:

```
    label = {
      name0 => value0,
      name1 => value1,
      …
    }
```

In other words, the order is not important and not guaranteed to remain by collectd.

If we were to add non-string label values in the future, would that influence the decision here?

Reply:
I 'm not sure that label set returns unique key or not.
If label set returns  `label[N-1]={name="foo", value="will be lost"}, label[N] = { name= "foo", vaule="something"}` in C world, in above example, it would be `label = { foo = "something" }` as outcome.

Reply:
Label keys are unique within a label set, so this overwrite behavior should not happen. I think that `label = { foo = "something", bar = "something else" }` is easier to work with.

Reply:
Thanks, I've got it.